common/src/components/snip/index.js

```diff
@@ -16,19 +16,22 @@ class Snip extends Component {
   }
 
   handleChange = (e) => {
-    const data = {
-      url: this.props.url,
-      note: e.target.value,
-    };
-    this.props.setNote(data);
-    this.props.updateStatus('syncing');
-    NoteService.save(data)
-      .then(() => {
-        this.props.updateStatus('saved');
-        setTimeout(() => this.props.updateStatus(''), 5000);
-      }).catch(() => {
-        this.props.updateStatus('error_saving');
-      });
+    if (this.saveTimeout) clearTimeout(this.saveTimeout);
+    this.saveTimeout = setTimeout(() => {
+      const data = {
+        url: this.props.url,
+        note: e.target.value,
+      };
+      this.props.setNote(data);
+      this.props.updateStatus('syncing');
+      NoteService.save(data)
+        .then(() => {
+          this.props.updateStatus('saved');
+          setTimeout(() => this.props.updateStatus(''), 5000);
```

Comment:
There is a possibility of race condition here. What if another event fired before this timeout is completed?

Reply:
I made a commit which resolves this. Please review it!